pkg/secureframe/graphql.go

```diff
@@ -572,166 +609,251 @@ func getCompanyTest(ctx context.Context, companyID string, token string, id stri
 		return Test{}, fmt.Errorf("API returned errors: %+v", out.Errors)
 	}
 
-	// log.Printf("out.Data: %+v", out.Data.Test)
+	log.Printf("out.Data: %+v", out.Data.Test)
 	return out.Data.Test, nil
 }
 
 func GetTests(ctx context.Context, companyID string, token string, reportKey string) ([]Test, error) {
 	log.Printf("Getting Secureframe tests for %s ...", reportKey)
-	tests, err := getCompanyTestV2s(ctx, companyID, token, reportKey)
-	if err != nil {
-		return nil, fmt.Errorf("get company test v2s: %w", err)
+
+	page := 0
+	totalPages := 1
+	maxRequests := 50
+	requestCount := 0
+
+	found := []Test{}
+	for page < totalPages {
+		page++
+
+		// Secondary protection to avoid accidentally DoS'ing Secureframe
+		requestCount++
+		if requestCount > maxRequests {
+			return found, fmt.Errorf("made too many requests")
+		}
+
+		ts, meta, err := getCompanyTestV2s(ctx, companyID, token, reportKey, page)
+		if err != nil {
+			return nil, fmt.Errorf("get company test v2s: %w", err)
+		}
+		found = append(found, ts...)
+		//time.Sleep(time.Duration(page) * time.Second)
+
+		totalPages = meta.TotalPages
+		page = meta.CurrentPage
 	}
 
-	log.Printf("got data on %d tests ... filling in", len(tests))
+	log.Printf("got data on %d tests ... filling in", len(found))
 	// The remaining bit of this function is a hack to fill in more information for failing tests.
 	// If we had a properly documented GraphQL API, we could get everything in a single query.
-	nts := []Test{}
-	for x, t := range tests {
+	detailed := []Test{}
+
+	for x, t := range found {
+		// No need for details in these cases
 		if t.Pass || !t.Enabled {
-			nts = append(nts, t)
+			detailed = append(detailed, t)
```

Comment:
The comment makes it seem like `continue` should just be used here. Is that correct?

Reply:
Good question. Not quite - "t" is the undetailed data struct that we append to the test list. We still need the undetailed data to know if we can close an existing issue.